modules/common/configmap/configmap.go

```diff
@@ -83,7 +96,7 @@ func createOrPatchConfigMap(
 		return "", op, fmt.Errorf("error create/updating configmap: %v", err)
 	}
 
-	configMapHash, err := util.ObjectHash(configMap)
```

Comment:
nice finding! I think it might be ok to just calc the hash on the configmap Data `configMapHash, err := util.ObjectHash(configMap.Data)` as this is usually the source we need to track for changes and e.g. restart resources if this information changes? Same applies for secrets I guess [1].

[1] https://github.com/openstack-k8s-operators/lib-common/blob/master/modules/common/secret/secret.go#L53

Reply:
We would need to hash at least `configMap.Data` and `configMap.BinaryData`.  I could create a specific structure to hold those 2 fields and hash them, but I don't know if that's actually clearer or better, because may at some point a consumer of a `ConfigMap` could be interested in detecting changes to the `ObjectMeta` field (I really don't know, so I wanted to err on the safe side).

Another solution would be to `Get` the `ConfigMap` from the OpenShift cluster to have the `TypeMeta` fields filled in by the cluster itself.

Reply:
right `configMap.Data` and `configMap.BinaryData` missed the second. I am just thinking if it makes sense for our case to hash the metadata. Lets say a new release of an operator adds another label to the CM template. If that change is not intended that the consumer of the CM to pick a different CM we don't want the pod to be recreated. So just hashing the actual data which is used by the consumer is probably what we want. If we decide/agree on that, we can just create a hash of the two hashes and return this, which is probably what you said (create structure to hold those two and hash them).

If we want to keep the has on the full CM, I'd vote for your other solution to just `Get` the CM after it got created. It won't required any future change if there needs to be any additional filed be set.

Reply:
Updated the patch using a struct to only hash `Data` and `BinaryData`.